### PR TITLE
Allows to display the death message only per player

### PR DESCRIPTION
**Pull request by Euphillya**
As done for advancements, I submit the same method for broadcast death message.

### CODE REVIEW

**Comment by granny** [comment]
Might as well combine this with the broadcast advancements patch I feel like

**Comment by katekerllenevich** [comment]
yes merge into the same patch please

**Comment by Euphillya** [comment]
its fine

File: patches/server/0206-Allows-change-broadcast-message-by-player.patch [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+@@ -1041,12 +1041,26 @@ public class EntityPlayer extends EntityHuman implements ICrafting {
+ 
+             if (scoreboardteambase != null && scoreboardteambase.getDeathMessageVisibility() != ScoreboardTeamBase.EnumNameTagVisibility.ALWAYS) {
+                 if (scoreboardteambase.getDeathMessageVisibility() == ScoreboardTeamBase.EnumNameTagVisibility.HIDE_FOR_OTHER_TEAMS) {
+-                    this.server.getPlayerList().a((EntityHuman) this, ichatbasecomponent);
++                    // Purpur Start - DeathMessage By Player
++                    if (net.pl3x.purpur.PurpurConfig.deathMessageOnlyBroadcastToAffectedPlayer) {
++                        this.sendMessage(ichatbasecomponent);
++                    } else {
++                        this.server.getPlayerList().a((EntityHuman) this, ichatbasecomponent);
```
Comment (katekerllenevich) [inline patches/server/0206-Allows-change-broadcast-message-by-player.patch]: OBFHELPER for `a`

File: patches/server/0206-Allows-change-broadcast-message-by-player.patch [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
++                    }
+                 } else if (scoreboardteambase.getDeathMessageVisibility() == ScoreboardTeamBase.EnumNameTagVisibility.HIDE_FOR_OWN_TEAM) {
+-                    this.server.getPlayerList().b(this, ichatbasecomponent);
++                    if (net.pl3x.purpur.PurpurConfig.deathMessageOnlyBroadcastToAffectedPlayer) {
++                        this.sendMessage(ichatbasecomponent);
++                    } else {
++                        this.server.getPlayerList().b(this, ichatbasecomponent);
```
Comment (katekerllenevich) [inline patches/server/0206-Allows-change-broadcast-message-by-player.patch]: OBFHELPER for `b`

**Review by katekerllenevich — CHANGES_REQUESTED** [review]

**Review by katekerllenevich — APPROVED** [review]

File: patches/server/0206-Allows-change-broadcast-message-by-player.patch [hunk 3]
```diff
@@ -33,17 +32,32 @@ index a1ee1066108985a95abddb03ff447b5a14f4f85f..4720644ae82f76f835f14c0b3a00e7b7
...
      private static void tpsCatchup() {
          tpsCatchup = getBoolean("settings.tps-catchup", tpsCatchup);
      }
 +
 +    public static boolean advancementOnlyBroadcastToAffectedPlayer = false;
-+    private static void advancementSettings() {
++    public static boolean deathMessageOnlyBroadcastToAffectedPlayer = false;
++    private static void broadcastSettings() {
 +        advancementOnlyBroadcastToAffectedPlayer  = getBoolean("settings.advancement.only-broadcast-to-affected-player", advancementOnlyBroadcastToAffectedPlayer );
++        deathMessageOnlyBroadcastToAffectedPlayer = getBoolean("settings.death.only-broadcast-to-affected-player", deathMessageOnlyBroadcastToAffectedPlayer);
```
Comment (BillyGalbreath) [inline patches/server/0206-Allows-change-broadcast-message-by-player.patch]: i think we should group these settings in the config like you did in code. `settings.broadcasts.whatever` or something. open to suggestions.

which means we'd also need a config version bump and a migration for the advancement setting

**Review by BillyGalbreath — CHANGES_REQUESTED** [review]

File: patches/server/0206-Allows-change-broadcast-message-by-player.patch (hunk 3 above)
Comment (Euphillya) [reply patches/server/0206-Allows-change-broadcast-message-by-player.patch]: Yes I think it will be better, I will do it right after in the next hour.

**Comment by Euphillya** [comment]
Its fine

**Review by BillyGalbreath — APPROVED** [review]

**Review by katekerllenevich — APPROVED** [review]

**Pull request merged**